compiler/compile/OMRSymbolReferenceTable.cpp

```diff
@@ -1023,6 +1039,14 @@ OMR::SymbolReferenceTable::findOrCreateSymRefWithKnownObject(TR::SymbolReference
    return result;
    }
 
+TR::SymbolReference *
+OMR::SymbolReferenceTable::createTempSymRefWithKnownObject(TR::Symbol *symbol, mcount_t owningMethodIndex, int32_t slot, TR::KnownObjectTable::Index knownObjectIndex)
```

Comment:
For non-temps we should start with the original symref and not the symbol - we'll need to be able to do the reverse and starting with the original symref will simplify changes to add this reverse capability in a follow-on change.

Reply:
yes, this one is for temps only. Another `findOrCreateSymRefWithKnownObject` is for non-temps which takes the original sym ref as first argument. I can change the name `findOrCreateSymRefWithKnownObject` to `findOrCreateNoTempSymRefWithKnownObject` to be more explicit.

Reply:
Can we add a fatal assert at least to check the symbol is a temp or something to stop someone using this by accident for a non-temp? The name does imply what you are saying, but I missed it reading the code so someone else might and a fatal assert would be the best thing if possible.

Reply:
Added assert to avoid confusion.